presto-main/src/main/java/com/facebook/presto/translator/registry/FunctionRegistry.java

```diff
@@ -0,0 +1,34 @@
+package com.facebook.presto.translator.registry;
+
+import com.facebook.presto.spi.function.FunctionMetadata;
+import com.google.common.collect.ImmutableMap;
+
+import java.lang.invoke.MethodHandle;
+import java.util.Arrays;
+
+import static com.facebook.presto.translator.registry.ScalarFromAnnotationsParser.parseFunctionDefinitions;
+
+public class FunctionRegistry
+{
+    private final ImmutableMap<FunctionMetadata, MethodHandle> translators;
+
+    public FunctionRegistry(Class<?>... classes)
```

Comment:
For registering MethodHandles to their corresponding function metadata so that they can be invoked via reflection.

Reply:
AbstractExpressionTranslator will use this class and ConstantTypeRegistry for it's default implementations